### PR TITLE
Update refined-github references

### PR DESCRIPTION
**Pull request by kidonng**
Related to refined-github/refined-github#4869

These links are referenced often (I believe) so saving some redirects would be awesome.

I also took the chance to rename `master` to `main` (fixed many invalid links as well).

I won't make pull requests to ask for the world to update the references, but this is one of the exceptions. Hope you don't mind that I'm going to mess up git blame 😄

### CODE REVIEW

File: .github/workflows/esm-lint.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ on:
       - '*'
   push:
     branches:
-      - master
+      - main
```
Comment (yakov116) [inline .github/workflows/esm-lint.yml:16]: is this duplicate now?

Comment (kidonng) [reply .github/workflows/esm-lint.yml:16]: Good catch, I initially manually removed it but then just redone everything with find and replace.

Comment (fregante) [reply .github/workflows/esm-lint.yml]: https://github.com/fregante/github-url-detection/blob/ca4edcfdc87e26241c10fb27985d73af8d024630/.github/workflows/esm-lint.yml#L5

**Comment by fregante** [comment]
These aren't real URLs for the most part, let's not mess up the blame 😃

**Pull request closed without merging**